luarules/gadgets/unit_area_timed_damage.lua

```diff
@@ -12,484 +12,229 @@ function gadget:GetInfo()
 end
 
 if not gadgetHandler:IsSyncedCode() then
-    return
+	return
 end
 
------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+---------------------------------------------------------------------------------------------------------------
+--
+-- Unit setup -- ! todo: Do this for all the units with area timed damage
+--
+-- (1) CustomParams. Add these properties to the unit (for ondeath) or weapon (for onhit).
+--
+-- area_timed_damage  -  true            -  Required.
+--    atd_duration    -  <number>        -  Required. The duration of the effect.
+--    atd_stable_ceg  -  <string> | nil  -  Name of the CEG that appears continuously for the duration.
+--    atd_damage_ceg  -  <string> | nil  -  Name of the CEG that appears on anything damaged by the effect.
+--    atd_resistance  -  <string> | nil  -  The 'type' of the damage, which can be resisted. Be uncreative.
+--
+-- (2) WeaponDefs. Add a new weapondef with its name set to:
+--		(a) the base weapon name plus "_area" for a specific weapon, and/or
+--		(b) "area_timed_damage" for an on-death hazard or any unmatched weapons; that is, if only the
+--			weapondef "area_timed_damage" is provided, it will be used both for any weapons without a
+--			matching name.."_area" weapondef and for the on-death explosion, if any.
+--
+--	Only a few properties of this wdef are used -- those that modify explosions:
+--    areaofeffect        -  Important.
+--    explosiongenerator  -  Important.
+--    impulse*            -  Important. Set to 0 in most cases. -- todo: might change
+--    weapontype          -  Important. Set to "Cannon" (or leave blank?) in most cases.
+--    damage              -  Important.
+--
+--	Any other properties that control projectile behaviors are ignored, in addition to:
+--    crater*             -  No effect.
+--    edgeeffectiveness   -  No effect. Set to 1.
+--    explosionspeed      -  No effect. Set to 10000.
+--
+---------------------------------------------------------------------------------------------------------------
 
--- ceg - ceg to spawn when explosion happens
--- damageCeg - ceg to spawn when damage is dealt
--- time - how long the effect should stay
--- damage - damage per second
--- range - from center to edge, in elmos
--- resistance - defines which units are resistant to this type of damage when it matches with 'areadamageresistance' customparameter in a unit.
+local WeaponAreaDefs  = {}
+local OnDeathAreaDefs = {}
+local TimedAreaDefs   = {}
```

Comment:
```suggestion
local weaponAreaDefs  = {}
local onDeathAreaDefs = {}
local timedAreaDefs   = {}
```
Locals should be lowercased